packages/experimental/Avatar/SPEC.md

```diff
@@ -0,0 +1,311 @@
+
+### Name
+
+Name prop is used to generate initials.
+
+### Shape
+
+The Avatar supports a circular and square (with rounded corners) shape.
+
+### Sizes
+
+`Avatar` sopports next sizes: `20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 96, 120`.
+Default size for win32 is 24.
+
+## API
+
+### Slots
+
+The `Avatar` component has the following slots:
+
+- `root` - The outer container representing the `Avatar` wrapper.
+- `image` - If specified, renders an `image`
+- `initials` - If specified, renders initials when there's no `image`
+- `icon` - If specified, renders an `icon` if there's no `image` or `initials`.
+- `initialsBackground` - It's a wrapper for initials and icon.
+- `ring` - If specified, renders the ring for the `Avatar` when active prop is set to `active`.
+- `badge` - If specified, renders presence badge. Example: `badge={{ status: 'outOfOffice' }}.`.
+
+The slots can be modified using the `compose` function on the `Avatar`. For more information on using the `compose` API, please see [this page](../../framework/composition/README.md).
+
+### Props
+
+```ts
+export interface JSAvatarProps extends IViewProps, AvatarConfigurableProps {
+  /**
+   * Optional activity indicator
+   * * active: the avatar will be decorated according to activeAppearance
+   * * inactive: the avatar will be reduced in size and partially transparent
+   * * unset: normal display
+   *
+   * @defaultvalue unset
+   */
+  active?: AvatarActive;
+
+  /**
+   * The appearance when `active="active"`
+   *
+   * @defaultvalue ring
+   */
+  activeAppearance?: AvatarActiveAppearance;
+
+  /**
+   * Badge shows the avatar's presence status.
+   * Badge can be shown only if `active` prop is undefined or unset
+   */
+  badge?: PresenceBadgeProps;
+
+  /**
+   * Icon to be displayed when the avatar doesn't have an image or initials.
+   *
+   * @defaultvalue `PersonRegular` (the default icon's size depends on the Avatar's size)
+   */
+  icon?: IconSourcesType;
+
+  /**
+   * Specify a string to be used instead of the name, to determine which color to use when color="colorful".
+   * Use this when a name is not available, but there is another unique identifier that can be used instead.
+   */
+  idForColor?: string | undefined;
+
+  /**
+   * The Avatar's image.
+   */
+  image?: ImageProps;
+
+  /**
+   * (optional) Custom initials.
+   *
+   * It is usually not necessary to specify custom initials; by default they will be derived from the `name` prop,
+   * using the `getInitials` function.
+   *
+   * The initials are displayed when there is no image (including while the image is loading).
+   */
+  initials?: string;
+
+  /**
+   * The name of the person or entity represented by this Avatar. This should always be provided if it is available.
+   *
+   * The name will be used to determine the initials displayed when there is no icon, as well as provided to
+   * accessibility tools.
+   */
+  name?: string;
```

Comment:
On Fluent Mobile and macOS, I've seen support for a "primary" and "secondary" text field. Primary might be the name, while secondary is the email. Is that not something that is in the web spec?

Reply:
I think that would be properties for an AvatarCard, not something that's for the Avatar. The Avatar itself would not have any text other than initials.